src/main/java/com/fasterxml/jackson/databind/jsontype/impl/AsArrayTypeDeserializer.java

```diff
@@ -146,6 +146,11 @@ protected String _locateTypeId(JsonParser p, DeserializationContext ctxt) throws
                             +ClassUtil.classNameOf(_idResolver)+") returned `null`");
                 }
                 return id;
+            }
+            // [databind#4849] Since 2.18.3, Should maybe allow defaultTyping without defaultImpl if type is already known
```

Comment:
@JooHyukKim Thinking this through, I don't think this is a proper fix, even tho it works around the specific failure here. I'll elaborate on the issue itself, but I think the problem may have been introduced by #4214 (in 2.17.0). Problem for tested case specifically is that customized `DefaultTypeResolverBuilder` forces addition of type information for EVERY type (for whatever reason), due to:

```
            @Override
            public boolean useForType(JavaType t) {
                return true;
            }
```

Or, it should. But while `EnumSetDeserializer` now expects polymorphic type id for `EnumSet` AND every `Enum` contained (... why is this desired?), I think `EnumSetSerializer` does not write said type id. And I guess it should.

Reply:
Ah okay. Let's close this for now. Will read through ur comment later.

Reply:
I committed my changes but it's very much work in progress. Too tired to continue for today, will pick up tomorrow :)